cmake/Findudunits.cmake

```diff
@@ -0,0 +1,57 @@
+# (C) Copyright 2022- UCAR.
+#
+# Try to find the udunits headers and library
+#
+# This module defines:
+#
+#   - udunits::udunits       - The udunits shared library and include directory, all in a single target.
```

Comment:
This file should maybe go into ESMA_CMake.  Need to discuss with @mathomp4

Reply:
Matt fixed.